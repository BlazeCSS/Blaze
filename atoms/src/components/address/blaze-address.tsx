import { Component } from '@stencil/core';

@Component({
  tag: 'blaze-address'
})
export class Address {

  render() {
    return (
      <address class="c-text c-address">
        <slot />
      </address>
    );
  }
}
