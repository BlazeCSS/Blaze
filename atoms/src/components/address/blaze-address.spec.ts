import { flush, render } from '@stencil/core/testing';
import { Address } from './blaze-address';
import { AddressHeading } from './blaze-address-heading';

describe('Address', () => {
  it('should build', () => {
    expect(new Address()).toBeTruthy();
    expect(new AddressHeading()).toBeTruthy();
  });

  describe('renders', () => {
    let element;

    const snapIt = (name, html) => {
      it(name, async () => {
        element = await render({
          components: [Address, AddressHeading],
          html
        });
        await flush(element);

        expect(element).toMatchSnapshot();
      });
    };

    snapIt(
      'correctly',
      '<blaze-address>Address line one</blaze-address>'
    );

    snapIt(
      'the address heading',
      `<blaze-address>
        <blaze-address-heading>Address header</blaze-address-heading>
        Address line one<br />
        Address line two<br />
        Postcode
      </blaze-address>`
    );
  });
});
