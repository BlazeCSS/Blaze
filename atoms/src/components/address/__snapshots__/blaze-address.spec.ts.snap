// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address renders correctly 1`] = `
<blaze-address
  class="hydrated"
>
  Address line one
  <address
    class="c-text c-address"
  />
</blaze-address>
`;

exports[`Address renders the address heading 1`] = `
<blaze-address
  class="hydrated"
>
  
        
  <blaze-address-heading
    class="hydrated"
  >
    Address header
    <span
      class="c-address__heading"
    />
  </blaze-address-heading>
  
        Address line one
  <br />
  
        Address line two
  <br />
  
        Postcode
      
  <address
    class="c-text c-address"
  />
</blaze-address>
`;
